internal/admission/admission.go

```diff
@@ -79,6 +82,23 @@ func credentialName(ctx context.Context, namespace, name string) string {
 	return fmt.Sprintf("%s-%s", prefix, randomStr)
 }
 
+func getGatewayNameFromCredentialName(credentialName []string) string {
+	const excludeNamespace = 1
```

Comment:
I don't think this will work for namespaces with a hyphen such as `atlas-data-lake`

I think another way is in `canMutateCredentialName`:
-  determine the index of the last `-` using `strings.LastIndex(currentCredentialName, "-")`. Then get substring of everything up to that point. That would be the `name-space-gateway-name` prefix.
- Take L71 that creates prefix into its own function.
- Use that function to generate the prefix for the input namespace, gatewayName.
- Return `prefix != substring`

Reply:
Yep, I completely forgot about a namespace with hyphens.  Good catch.